### PR TITLE
Remove retry from "Then" steps.

### PR DESCRIPTION
**Pull request by Chekote**
# Problem:
The waitFor functionality is implemented inconsistently. It should be applied uniformly to all "Then" steps.

Even if it is applied uniformly to all "Then" steps, it is still incumbent upon the developer to add the functionality to all new "Then" steps, and also incumbent upon the FlexibleMink consumer to add the functionality to their own "Then" steps. We would ideally like to remove the need for this repetitive plumbing.

# Implementation:
I wrote a Behat extension called [BehatRetryExtension](https://github.com/Chekote/BehatRetryExtension). This can be added to any project using Behat, and will automatically retry all "Then" steps that are attempted while the extension is active.

I initially added this extension to FlexibleMink, but after discussion with the team, we decided that the retry functionality should be left out of FlexibleMink, and the consumer of the FlexibleMink package can decide if they wish to bring in the BehatRetryExtension also.

### CODE REVIEW

**Comment by medology-bugman** [comment]
Paging @atroppe to review the front-end work in this pull request!
Paging @PrinceMegahit to review the back-end work in this pull request!
Paging @jose-ochoa to review the testing work in this pull request!

I couldn't recognize who these files were relevant to:
composer.lock

**Comment by Chekote** [comment]
!reassign front-end

**Comment by medology-bugman** [comment]
Paging @mehtaamish89 to review the front-end work in this pull request!

File: src/Medology/Behat/Mink/FlexibleContext.php [hunk 1]
```diff
@@ -36,36 +35,6 @@ class FlexibleContext extends MinkContext
...
-    public function assertPageAddress($page)
-    {
-        Spinner::waitFor(function () use ($page) {
-            parent::assertPageAddress($page);
-        });
-    }
-
     /**
      * This method overrides the MinkContext::assertPageContainsText() default behavior for assertPageContainsText to
      * ensure that it waits for the text to be available with a max time limit.
```
Comment (jose-ochoa) [inline src/Medology/Behat/Mink/FlexibleContext.php]: Should the description be updated?

- [ ] Addressed
- [ ] Approved

Comment (Chekote) [reply src/Medology/Behat/Mink/FlexibleContext.php]: Yes. Good catch. I've updated all docblocks that mention waiting.

**Pull request merged**